jsettlers.logic/src/main/java/jsettlers/logic/buildings/trading/TradingBuilding.java

```diff
@@ -174,4 +220,47 @@ private void drawWaypointLine(boolean draw) {
 	public EPriority[] getSupportedPriorities() {
 		return SUPPORTED_PRIORITIES;
 	}
+
+	public boolean setDock(int[] position) {
+		if (this.type != EBuildingType.HARBOR) {
+			return false;
+		}
+		if (this.dockPosition != null) { // replace dock
+			this.grid.setDock(this.dockPosition, false, this.getPlayer());
+		}
+		this.dockPosition = position;
+		this.grid.setDock(position, true, this.getPlayer());
+		this.shipWayStart = new ShortPoint2D
```

Comment:
`EDirection` can do this / something similar.

`getNextHexPoint(ShortPoint2D pos, int steps)`